apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TestDatabaseFixture.cs

```diff
@@ -1,68 +1,78 @@
 {
-    public class TestDatabaseFixture : IDisposable {
-
-        private readonly string _databaseName = "t_" + Guid.NewGuid().ToString("N").Substring(0, 28);
-        private Task _creationTask;
+    public class TestDatabaseFixture : IDisposable
+    {
+        // scratch can be used to run tests on a precreated db.
+        // all tests are designed to be re-run on an exiting db (previously written state will
+        // not cause failures).
+        // if you use a scratch database, be sure to comment out the database
+        // creation and disposal methods.
+        private readonly string _databaseName = //"scratch";  
+            "t_" + Guid.NewGuid().ToString("N").Substring(0, 28);
         private readonly string _testTable = "TestTable";
+        private Task _creationTask;
         private DatabaseAdminClient _databaseAdminClient;
 
-        public void Dispose() {
-            if (_databaseAdminClient != null) {
-                _databaseAdminClient.DropDatabase(new DatabaseName(TestProjectName, TestInstanceName, DatabaseName));
-            }
-        }
-
         public string TestInstanceName => "myspanner";
 
         public string TestProjectName => "spanneref";
 
-        public string ConnectionString => "Data Source=" + TestProjectName+ "/" + TestInstanceName+ "/"
-                                          + DatabaseName;
+        public string ConnectionString => "Data Source=" + TestProjectName + "/" + TestInstanceName
+                                          + "/" + DatabaseName;
 
         public string DatabaseName => _databaseName;
 
         public string TestTable => _testTable;
 
-        public async Task<SpannerConnection> GetTestDatabaseConnectionAsync() {
+        public void Dispose()
+        {
+            if (_databaseAdminClient != null)
+            {
+                _databaseAdminClient.DropDatabase(
+                    new DatabaseName(TestProjectName, TestInstanceName, DatabaseName));
+            }
+        }
+
+        public async Task<SpannerConnection> GetTestDatabaseConnectionAsync()
+        {
+            await EnsureTestDatabaseAsync();
+            return new SpannerConnection(ConnectionString);
+        }
+
+        public async Task EnsureTestDatabaseAsync()
+        {
             TaskCompletionSource<int> creationTaskCompletionSource;
-            if (_creationTask == null) {
+            if (_creationTask == null)
+            {
                 creationTaskCompletionSource = new TaskCompletionSource<int>();
                 if (Interlocked.CompareExchange(ref _creationTask,
                         creationTaskCompletionSource.Task,
-                        null) != null) {
+                        null) != null)
+                {
                     await _creationTask;
-                    return new SpannerConnection(ConnectionString);
+                    return;
                 }
-            } else {
+            } else
```

Comment:
previous code was also wrong -- so this issue exists throughout the codebase.  so I need to do a full pass later.